qualname.py

```diff
@@ -47,10 +47,9 @@ def qualname(obj):
     # class/method/function is defined.
     try:
         filename = inspect.getsourcefile(obj)
-    except TypeError:
-        return obj.__qualname__  # raises a sensible error
-    if not filename:
-        return obj.__qualname__  # raises a sensible error
```

Comment:
I'm pretty sure this part is still important too.

Reply:
All the tests passed, do you know what case actually hits this branch?

I don't know of when `getsourcefile` returns something falsy, but probably can read through the source to see.